### DISCUSSION TITLE
Nuxt 3 on generate command generate html with meta refresh tag only and not the actual content

### DISCUSSION THREAD

**Discussion by razahaider9400** [nuxt/nuxt#27462] [Questions]
``When I run the npm run generate command it generate the payload and html for each route which is working fine. but for the blog page It only generate the following html

`<!DOCTYPE html><html><head><meta http-equiv="refresh" content="0; url=/"></head></html>`

and there is no payload file. it is working fine on local.

Any idea or solution ?

Thanks

I haven't found any data on this yet. Here is the code

`<template>

    <main class="page_content">
        <section class="page_banner" style="
                background-image: url('/assets/images/shapes/tyre_print_3.svg');
            ">
            <div class="container">
                <ul class="breadcrumb_nav unordered_list">
                    <li>
                        <router-link to="/">
                            <a href="#">Home</a>
                        </router-link>
                    </li>
                    <li>
                        <a href="#">Blogs</a>
                    </li>
                </ul>
                <h1 class="page_title wow" data-splitting>Blogs</h1>
            </div>
        </section>
        <div class="my-5 text-center">
            <h3>Blogs</h3>
        </div>
        <div v-if="!loader">
            <section class="details_section pb-5 mb-5">
                <div class="container">
                    <div class="row">
                        <div class="col-lg-4 col-md-6" v-for="blog in blogs.data" :key="blog.id">
                            <NuxtLink :to="{
            name: 'blog-slug',
            params: { slug: blog.slug }
        }">
                                <div class="card" style="border-radius: 35px">
                                    <img :src="'https://winkcardetailing.com.au/Uploads/Blog/' + blog.image"
                                        alt="Blog Image" style="
                                        height: 300px;
                                        width: 500px;
                                        border-radius: 35px;
                                    " />
                                    <div class="card-body">
                                        <p class="card-text h4 text-center">
                                            {{
            blog.title.length > 30
                ? blog.title
                    .slice(0, 30)
                    .trim() + ".."
                : blog.title
                                            }}
                                        </p>
                                    </div>
                                </div>
                            </NuxtLink>
                        </div>
                        <div class="mt-4">
                            <Bootstrap5Pagination :data="blogs" size="large" align="center" class="flex-wrap"
                                @pagination-change-page="getBlog" />
                        </div>
                    </div>
                </div>
            </section>
        </div>
        <div v-else>
            <Loader />
        </div>
    </main>
</template>
<script setup>
import { Bootstrap5Pagination } from "laravel-vue-pagination";
import Loader from "../../components/Loader.vue";
import { ref, onMounted, defineComponent } from 'vue';

const blogs = ref([]);
const loader = ref(true);
const page = ref(1);

defineComponent(() => {
    Loader,
        Bootstrap5Pagination
});

getBlog();

async function getBlog(pages) {

    if (pages) {
        page.value = pages;
    }
    const { data: getBlogs } = await useMyFetch("/api/blogs?page=" + page.value, {
        method: "GET",
    });
    if (getBlogs.value) {
        blogs.value = getBlogs.value;
        loader.value = false;
    }

}
onMounted(() => {
    window.scrollTo({ top: 0, behavior: "smooth" });
})
</script>`

**Comment by xjccc** [comment]
like this?
```
async function getBlog(pages) {
  if (pages) {
      page.value = pages;
  }
  const { data: getBlogs } = await useMyFetch("/api/blogs?page=" + page.value, {
      method: "GET",
  });
  if (getBlogs.value) {
      blogs.value = getBlogs.value;
      loader.value = false;
  }
}
await getBlog();
```

**Reply by razahaider9400** [reply]
This was not the issue
I found the bug actullay i am using the third party paginatiion package which was causing the issue when i run npm run generate. I guess when the crawler visit the page it break the page and that's why it was putting the meta refresh tag to redirect to home page because the page is broken. I wrapped that componenet in `<ClientOnly></ClientOnly>` and now it is generating the payload for it and proper html rather than that meta refresh tag.
